Find a functionally equivalent code:android.animation.ObjectAnimator.animateValue:COMMENT
Method Modifier: 
Comment:/**
 * This method is called with the elapsed fraction of the animation during every
 * animation frame. This function turns the elapsed fraction into an interpolated fraction
 * and then into an animated value (from the evaluator. The function is called mostly during
 * animation updates, but it is also called when the <code>end()</code>
 * function is called, to set the final value on the property.
 *
 * <p>Overrides of this method must call the superclass to perform the calculation
 * of the animated value.</p>
 *
 * @param fraction The elapsed fraction of the animation.
 */

@@ -1,7 +1,13 @@
 {
+    final Object target = getTarget();
+    if (mTarget != null && target == null) {
+        // We lost the target reference, cancel and clean up.
+        cancel();
+        return;
+    }
     super.animateValue(fraction);
     int numValues = mValues.length;
     for (int i = 0; i < numValues; ++i) {
-        mValues[i].setAnimatedValue(mTarget);
+        mValues[i].setAnimatedValue(target);
     }
 }

